src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -1222,4 +1223,1592 @@ struct S3
+            //     public DerivedDerived1()
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived1").WithArguments("Derived").WithLocation(8, 12),
+            // (12,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the 'SetsRequiredMembers' attribute.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (12,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the 'SetsRequiredMembers' attribute.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (12,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived2 : Derived
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived2").WithArguments("Derived").WithLocation(12, 8),
+            // (15,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the 'SetsRequiredMembers' attribute.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8),
+            // (15,8): error CS9509: The required members list for the base type 'Derived' is malformed and cannot be interpreted. To use this constructor, apply the 'SetsRequiredMembers' attribute.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersBaseTypeInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8),
+            // (15,8): error CS9508: The required members list for 'Derived' is malformed and cannot be interpreted.
+            // record DerivedDerived3() : Derived;
+            Diagnostic(ErrorCode.ERR_RequiredMembersInvalid, "DerivedDerived3").WithArguments("Derived").WithLocation(15, 8)
+        );
+    }
+
+    [Fact]
+    public void EnforcedRequiredMembers_ShadowedInSource_06()
+    {
+        var original = @"
+public record Base
+{
+    public required int P { get; init; }
+}
+";
+
+        var originalComp = CreateCompilationWithRequiredMembers(new[] { original, IsExternalInitTypeDefinition }, assemblyName: "original");
+        CompileAndVerify(originalComp, verify: ExecutionConditionUtil.IsCoreClr ? Verification.Passes : Verification.Skipped).VerifyDiagnostics();
+
+        // This IL is the equivalent of:
+        // public record Derived : Base
```

Comment:
nit: For these `CompileIL`, I don't see the point of bloating the IL by making it a record. A class would have sufficed to the purpose of those tests.

Reply:
It would not. The point of these tests is for copy constructors and similar, and in a future PR I'm working on these being records pointed out some considerations I'll need to think about for how copy constructors work with `SetsRequiredMembers`. I'll keep these as is.